src/main/java/org/testng/internal/TestResult.java

```diff
@@ -33,33 +38,49 @@
   private int parameterIndex;
   private boolean m_wasRetried;
 
-  public TestResult() {}
+  private TestResult() {
+    //defeat instantiation. We have factory methods.
+  }
+
+  public static TestResult newEmptyTestResult() {
+    return new TestResult();
+  }
+
+  public static TestResult newTestResultFor(ITestNGMethod method) {
+    return newContextAwareTestResult(method, null);
+  }
 
-  public TestResult(ITestNGMethod method, Throwable throwable, ITestContext context) {
+  public static TestResult newContextAwareTestResult(ITestNGMethod method, ITestContext ctx) {
+    TestResult result = newEmptyTestResult();
     long time = System.currentTimeMillis();
-    init(method.getTestClass(), method, throwable, time, time, context);
+    result.init(method, null, time, 0L, ctx);
+    return result;
   }
 
-  public TestResult(
-      IClass testClass,
-      ITestNGMethod method,
-      Throwable throwable,
-      long start,
-      long end,
-      ITestContext context) {
-    init(testClass, method, throwable, start, end, context);
+  public static TestResult newTestResultWithCauseAs(ITestNGMethod method, Throwable t,

```

Comment:
Try to keep the same param order: `method, context, throwable, start`

Reply:
Fixed